src/services/TestResultsService.ts

```diff
@@ -302,13 +302,13 @@ export class TestResultsService {
                     const firstTestAfterAnvCompareDate = dateFns.addYears(dateFns.startOfMonth(regOrFirstUseDate!), 1);
                     // Checks for testType = First test or First test Retest AND test date is 1 year from the month of first use or registration date
                     if (this.isFirstTestRetestTestType(testType) && dateFns.isAfter(new Date(), firstTestAfterAnvCompareDate)) {
-                        testType.testExpiryDate = dateFns.addYears(dateFns.lastDayOfMonth(new Date()), 1).toISOString();
+                      testType.testExpiryDate = dateFns.lastDayOfMonth(dateFns.addYears(new Date(), 1)).toISOString();
                     } else if (this.isFirstTestRetestTestType(testType) && dateFns.isEqual(mostRecentExpiryDateOnAllTestTypesByVin, new Date(1970, 1, 1))) {
-                      const anvDateForCompare = regOrFirstUseDate ? dateFns.addYears(dateFns.lastDayOfMonth(regOrFirstUseDate), 1).toISOString() : undefined;
+                      const anvDateForCompare = regOrFirstUseDate ? dateFns.addYears(dateFns.endOfDay(dateFns.lastDayOfMonth(regOrFirstUseDate)), 1).toISOString() : undefined;
```

Comment:
Don't you have to switch the order of operations here also to handle the leap years?

Reply:
done